lib/puppet/provider/git/default.rb

```diff
@@ -16,6 +16,8 @@ def run_cwd(dir=resource[:path], &block)
 
   def commit
     run_cwd{ git('rev-parse', '--verify', resource[:commit]) }
+    head = run_cwd{ git('rev-parse', 'HEAD') }
```

Comment:
There is no need to set and return the head variable. Just the run_cwd will be fine.